lib/web_ui/dev/build.dart

```diff
@@ -115,22 +115,29 @@ class GnPipelineStep extends ProcessStep {
 ///
 /// Can be safely interrupted.
 class NinjaPipelineStep extends ProcessStep {
-  NinjaPipelineStep({required this.target});
+  NinjaPipelineStep({required this.host});
 
   @override
   String get description => 'ninja';
 
   @override
   bool get isSafeToInterrupt => true;
 
+  final bool host;
+
   /// The target directory to build.
-  final Directory target;
+  Directory get target =>
+      host ? environment.hostDebugUnoptDir : environment.wasmReleaseOutDir;
 
   @override
   Future<ProcessManager> createProcess() {
     print('Running autoninja...');
     return startProcess(
-      'autoninja',
+      // When build the wasm target that includes CanvasKit, we need to use
+      // `ninja` directly to avoid having a high `-j` value that could
+      // potentially slow down the entire machine because emscripten doesn't use
+      // goma.
+      host ? 'autoninja' : 'ninja',
```

Comment:
```dart
'autoninja',
if (!host)
  '--offline',
```

may do the trick? That instructs autoninja to *not* look for goma, so it'll default to whatever J it deems good for local builds.